datafusion/optimizer/Cargo.toml

```diff
@@ -44,6 +45,7 @@ async-trait = { workspace = true }
 chrono = { workspace = true }
 datafusion-common = { workspace = true, default-features = true }
 datafusion-expr = { workspace = true }
+datafusion-functions-array = { workspace = true, optional = true }
```

Comment:
@alamb I think having optional importing is fine. Is there any other downside of this?